library/systemd/test/yast2/system_service_test.rb

```diff
@@ -458,137 +458,43 @@
   end
 
   describe "#start" do
+    let(:service_active) { false }
+    let(:socket_active) { false }
+
     it "sets the action to :start" do
       system_service.start
 
       expect(system_service.action).to eq(:start)
     end
 
-    context "when the service is not active" do
-      let(:service_active) { false }
-
-      context "and the socket is not active" do
-        let(:socket_active) { false }
-
-        it "sets the service to be active" do
-          expect { system_service.start }.to change { system_service.active? }
-            .from(false).to(true)
-        end
-
-        it "sets the service as changed" do
-          expect { system_service.start }.to change { system_service.changed?(:active) }
-            .from(false).to(true)
-        end
-      end
-
-      context "and the socket is active" do
-        it "sets the service to stay as active" do
-          expect { system_service.start }.to_not change { system_service.active? }
-        end
-
-        it "ignores the change" do
-          system_service.start
-          expect(system_service.changed?).to eq(false)
-        end
-      end
+    it "sets the service to be active" do
+      expect { system_service.start }.to change { system_service.active? }
+        .from(false).to(true)
     end
 
-    context "when the service is already active (started)" do
-      let(:service_active) { true }
-
-      it "sets the service to stay as active" do
-        expect { system_service.start }.to_not change { system_service.active? }
-      end
-
-      it "ignores the change" do
-        system_service.start
-        expect(system_service.changed?).to eq(false)
-      end
-    end
-
-    context "and the service was set to be stopped previously" do
-      before do
-        system_service.stop
-      end
-
-      it "sets the service to stay as active" do
-        expect { system_service.start }.to change { system_service.active? }
-          .from(false).to(true)
-      end
-
-      it "cancels the previous change" do
-        system_service.start
-        expect(system_service.changed?).to eq(false)
-      end
+    it "sets the service as changed" do
```

Comment:
This remembers me that we needs to adapt Services Manager, see https://github.com/yast/yast-services-manager/blob/SLE-15-GA/src/lib/services-manager/widgets/services_table.rb#L282

Reply:
I've reimplemented the changes calculation mechanism so it should work out-of-the-box.